docs/Configuration.rst

```diff
@@ -5981,7 +6036,166 @@ data is stored in the database.
+
+      -  Cheap transaction: this option will use the gas price configured by the following property (in `WEI <http://ethdocs.org/en/latest/ether.html>`__):
+
+      .. code-block:: jproperties
+
+            exo.wallet.transaction.gas.cheapPrice=4000000000
+
+      -  Normal transaction: this option will use the gas price configured by the following property (in `WEI <http://ethdocs.org/en/latest/ether.html>`__):
+
+      .. code-block:: jproperties
+
+            exo.wallet.transaction.gas.normalPrice=8000000000
+
+      -  Fast transaction: this option will use the gas price configured by the following property (in `WEI <http://ethdocs.org/en/latest/ether.html>`__):
+
+      .. code-block:: jproperties
+
+            exo.wallet.transaction.gas.fastPrice=15000000000
+
+
+* When a transaction is sent and wasn't mined for several days, it will be marked as failed in eXo Wallet internal database.
+  The maximum number of days waiting for transaction mining can be configured using the following property:
+
+    .. code-block:: jproperties
+
+          exo.wallet.transaction.pending.maxDays=3
+
+
+    .. note:: If the transaction still not mined after ``exo.wallet.transaction.pending.maxDays`` days, it will be marked as failed transaction in internal eXo database.
+              When it gets mined, the Job ``ContractTransactionVerifierJob`` will detect it and will store again the real transaction status coming from blockchain and send notifications.
+
+
+Wallet types
+~~~~~~~~~~~~
+
+* **Admin wallet**
+
+  The *Admin wallet* is created automatically during first startup of the eXo Platform server.
+
+  In order to be able to issue transactions on the blockchain using `Web3j<https://web3j.io/>`__ server side, the private key of the *Admin wallet* is stored (in encrypted form) in the internal database.
+  The private key is encrypted by combining two passwords:
+
+     -  A first password is read from a keystore file ( see :ref:`Updating password encryption key <UpdatePasswordEncryptionKey>` )
+
+     .. warning:: The generated keystore file under `gatein/conf/codec` MUST be backed up as a data folder, because it contains a key file that is used to decrypt stored wallets private keys. If it's lost, all wallets private keys will be lost and consequently, all funds would be lost and unrecoverable.
+
+     -  A second password that can be configured in properties:
+
+    .. code-block:: jproperties
+
+          exo.wallet.admin.key=changeThisKey
+
+
+    .. warning:: The password can't be changed once the platform is started for the first time. In fact, this password will be used to encrypt the *Admin wallet*'s private key that will be stored in database.
+        If its value is modified after server startup, the private key of admin wallet won't be decrypted anymore, preventing all administrative operations.
```

Comment:
If an admin does this mistake (modifies after first startup), is the operation destrictive or a simple roll back to the previous value will let the admin operations work again ?

Also : is there a way to deliberately reset (e.g in case of compromision) the admin wallet private key ? Can this be done just by wiping out both  the key and the key store ?

Reply:
That's right, we must give the possibility to change the password. It can be done using a UI that I deleted (Admin account in Wallet admin). So unfortunately, for now, we don't have a solution to reset this password. (May be to add in the list of User stories of Feature, WDYT ?)